sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/b2c/validator/AADB2CJwtAppIdValidator.java

```diff
@@ -0,0 +1,43 @@
+package com.azure.spring.aad.b2c.validator;
+
+import com.azure.spring.aad.webapi.validator.AADJwtClaimValidator;
+import com.azure.spring.autoconfigure.aad.AADTokenClaim;
+import java.util.Set;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.core.OAuth2TokenValidatorResult;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.util.Assert;
+
+/**
+ * Validates the "azp" claim in a {@link Jwt}, that is matches a configured value.
+ */
+public class AADB2CJwtAppIdValidator implements OAuth2TokenValidator<Jwt> {
+
+    private final AADJwtClaimValidator<Set<String>> validator;
+
+    /**
+     * Constructs a {@link AADB2CJwtAppIdValidator} using the provided parameters
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public AADB2CJwtAppIdValidator(Set<String> aclLists) {
+        if (!aclLists.isEmpty()) {
+            this.validator = new AADJwtClaimValidator<>(AADTokenClaim.APPID, appId -> {
+                if (appId == null) {
+                    return true;
+                }
+                return aclLists.contains(appId);
+            });
+        } else {
+            this.validator = new AADJwtClaimValidator<>(AADTokenClaim.APPID, appId -> true);
+        }
+    }
```

Comment:
can we move the ` _if (!aclLists.isEmpty()) {_ `  statement into the Predicate logic？
![image](https://user-images.githubusercontent.com/4465723/105467190-4d041c00-5cd0-11eb-991c-fca8d35e1c03.png)